### PR TITLE
Error on document write failure

### PR DESCRIPTION
**Pull request by alexandercampbell-wf**
### Code Review

Report failures that occur during document generation.

I also fixed all the type-check failures caused by the latest stretchr/testify updates.

### Reviewers

@tylertreat-wf @aaronkavlie-wf @stevenosborne-wf

### CODE REVIEW

File: rest/api.go [hunk 1]
```diff
@@ -225,7 +225,9 @@ func (r *muxAPI) StartTLS(addr Address, certFile, keyFile FilePath, middleware .
 func (r *muxAPI) preprocess() {
 	r.validateRulesOrPanic()
 	if r.config.GenerateDocs {
-		newDocGenerator().generateDocs(r)
+		if err := newDocGenerator().generateDocs(r); err != nil {
+			log.Fatal(err)
+		}
```
Comment (alexandercampbell-wf) [inline rest/api.go]: This is the primary change. I noticed that the `_docs` generation would sometimes silently fail and found the culprit was this line.

If you prefer, I could make this `log.Println()` instead? @tylertreat-wf

**Comment by tylertreat-wf** [comment]
+1

File: rest/api.go (hunk 1 above)
Comment (tylertreat-wf) [inline rest/api.go]: Yeah, `Println` might be preferable.

Comment (alexandercampbell-wf) [inline rest/api.go]: Done.

**Comment by tylertreat-wf** [comment]
+1

**Pull request merged**